### PR TITLE
Purchase Management: Display introductory offer information when relevant

### PR DESCRIPTION
**Pull request by danielmx-dev**
#### Changes proposed in this Pull Request

* Adds new messages to the purchase list and purchase detail screens when an introductory offer is active. Feedback is welcome about what is the best way (and language) to communicate this.

In Purchase List adds changes the regular message (only for free trials):

**When auto-renew is on**
image

**When auto-renew is off**
image

In the purchase details it reuses the same message implemented for checkout in #50494
image

#### Testing instructions

Follow the steps in D59006-code to purchase a subscription with an introductory offer.
Open the purchases list and verify that the subscription is explicit about the free trial it contains.
Open the purchase detail and verify that it includes the duration of the free trial right below the price.

### CODE REVIEW

File: client/me/purchases/purchase-item/index.jsx [hunk 1]
```diff
@@ -100,6 +102,38 @@ class PurchaseItem extends Component {
 			);
 		}
 
+		if ( isWithinIntroductoryOfferPeriod( purchase ) && isIntroductoryOfferFreeTrial( purchase ) ) {
+			if ( isRenewing( purchase ) ) {
+				return translate( 'Free trial ends on {{span}}%(date)s{{/span}}, renews automatically', {
+					args: {
+						timeUntilExpiry: expiry.fromNow(),
```
Comment (daledupreez) [inline client/me/purchases/purchase-item/index.jsx]: I don't think `timeUntilExpiry` is used here.

**Review by daledupreez — COMMENTED** [review]
From a wording and presentation perspective, these changes look good to me.

I haven't had a chance to test this extensively, but can probably do so tomorrow.

**Comment by sirbrillig** [comment]
I haven't had a chance to review this in detail yet, but I wanted to mention that we should keep in mind the changes in https://github.com/Automattic/wp-calypso/pull/51198 which I think touch these same files. I'll comment there too.

**Review by sirbrillig — APPROVED** [review]
I have not tested this, but from a code perspective, this looks good to me.

**Review by chrismccluskey — APPROVED** [review]
I've tested this along with D59082-code and everything works as expected.

File: client/me/purchases/purchase-item/index.jsx [hunk 2]
```diff
@@ -100,6 +102,37 @@ class PurchaseItem extends Component {
 			);
 		}
 
+		if ( isWithinIntroductoryOfferPeriod( purchase ) && isIntroductoryOfferFreeTrial( purchase ) ) {
+			if ( isRenewing( purchase ) ) {
+				return translate( 'Free trial ends on {{span}}%(date)s{{/span}}, renews automatically', {
```
Comment (michaeldcain) [inline client/me/purchases/purchase-item/index.jsx]: To be a little more in line with the current renewal status, can we change this to `Free trial ends on {{span}}%(date)s{{/span}}, renews automatically at %(amount)s`?

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5614551

Thank you @dylanrevisited for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5614551).

**Pull request merged**